assets/js/modules/analytics-4/datastore/settings.js

```diff
@@ -98,6 +98,10 @@ export function validateCanSubmitChanges( select ) {
 	invariant( haveSettingsChanged(), INVARIANT_SETTINGS_NOT_CHANGED );
 	invariant( ! isDoingSubmitChanges(), INVARIANT_DOING_SUBMIT_CHANGES );
 
-	invariant( isValidPropertySelection( getPropertyID() ), INVARIANT_INVALID_PROPERTY_SELECTION );
-	invariant( isValidWebDataStreamID( getWebDataStreamID() ), INVARIANT_INVALID_WEBDATASTREAM_ID );
+	const propertyID = getPropertyID();
+	invariant( isValidPropertySelection( propertyID ), INVARIANT_INVALID_PROPERTY_SELECTION );
+	invariant(
+		propertyID === PROPERTY_CREATE || isValidWebDataStreamID( getWebDataStreamID() ),
+		INVARIANT_INVALID_WEBDATASTREAM_ID,
+	);
```

Comment:
Let's make this `invariant` call conditionally based on the `propertyID` rather than making that condition part of the invariant logic for a valid webdatastream (i.e. move property condition to wrapping `if`).